.editorconfig

```diff
@@ -243,12 +282,21 @@ dotnet_diagnostic.IDE0100.severity = warning
 # Simplify LINQ expression.
 dotnet_diagnostic.IDE0120.severity = warning
 
+# Use block-scoped namespace.
+dotnet_diagnostic.IDE0160.severity = warning
+
+# Use file-scoped namespace.
+dotnet_diagnostic.IDE0161.severity = warning
```

Comment:
The above one tells it to use block-scope and this tells it to use file-scope. How does this work?

Reply:
Yeah some of these rules go against each other.

Reply:
These two rules work in tandem with an option - which I have left implicit as the default works for us.
`csharp_style_namespace_declarations = block_scoped` - based on the option only one of the rules will ever enforce things.
So having both rules enabled isn't strictly required - but it's less faff down the line if we want to change the option - you don't also have to hunt up a corresponding rule - we just tweak the option and it does the job. Really it would've made more sense if they'd just shipped a single rule to enforce this.
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0160-ide0161

In a later PR I'd like to reformat the file to something like this - which groups the options and their rules together and fixes the disconnect we currently have of keeping the options and rules in seperate groups.
```editorconfig
# IDE0160/IDE0161 Use block-scoped namespace/Use file-scoped namespace
csharp_style_namespace_declarations = block_scoped
dotnet_diagnostic.IDE0160.severity = warning
dotnet_diagnostic.IDE0161.severity = warning
```

So yes, enabling both rules is intentional, desired, and won't break things.

Reply:
> In a later PR I'd like to reformat the file to something like this

Perhaps this should be part of this PR? In first commit reformat then in the second add these additional rules

Reply:
I guess it is still weird to list rules we will never enable, but I see why now. Doing both in one commit in this PR should be fine imo.

Reply:
I'd like to reformat the whole section, not just these specific rules - a follow-up PR would be much cleaner.

Reply:
I'm thinking this is all should be part of the same PR

Reply:
I have pushed a second commit with the reformat, but let me know if it's too much for the one PR.